### PR TITLE
Run migrations in order across repos

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR makes [`/entrypoint.sh db migrate`](https://github.com/plausible/community-edition/blob/4e949e84f787702835ec0654c7f663dc2fb61dbe/docker-compose.yml#L28C63-L28C88) and `/entrypoint.sh db pending-migrations` order all pending migrations and run them in a single queue instead of per-repo.

Related card: https://3.basecamp.com/5308029/buckets/29267832/card_tables/cards/7508896723

<details><summary>Contents of that card</summary>

<br/>

> Some migrations in PostgreSQL implicitly depend on migrations in ClickHouse (there is at least one such migration), and because PostgreSQL migrations run first, they fail because from their point of view ClickHouse schemas are now "outdated":
> 
> How migrations were added (example):
> - add table to PostgreSQL db
> - add column to ClickHouse table
> - run PostgreSQL data migration that reads from new column in ClickHouse
> 
> How migrations are run:
> - run all Plausible.Repo migrations
> - run all Plausible.IngestRepo migrations
> 
> Which results in this broken order:
> - add table to PostgreSQL db
> - run PostgreSQL data migration that reads from new column in ClickHouse <- fails
> - add column to ClickHouse table <- never runs
> 
> It might be possible to run them in a single queue / order in `db migrate` command:
> - collect all migrations to be run, order them by version / date
> - start all repos (Repo, IngestRepo, ClickhouseRepo)
> - execute them one by one

</details>

### Tests
- [x] Automated tests have been added (only for pending migrations listings, I'll try to add an actual `db migrate` test later)

### Changelog
- [x] Entry has been added to changelog

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible_release.ex [hunk 1]
```diff
@@ -25,6 +25,54 @@ defmodule Plausible.Release do
     IO.puts("Migrations successful!")
   end
 
+  # Unlike `migrate/0` above this function:
```
Comment (ukutaht) [inline lib/plausible_release.ex:28]: I think we should add documentation for why we run migrations differently from the 'default' way.

Comment (ruslandoga) [reply lib/plausible_release.ex:28]: Done in https://github.com/plausible/analytics/pull/4466/commits/818401c4266f2beb6a6a1ce1f9bac12624d36dbb

Comment (ruslandoga) [reply lib/plausible_release.ex]: Done in https://github.com/plausible/analytics/pull/4466/commits/3d2e060c991fcdafb713dea8432040de1cf520cb

File: lib/plausible_release.ex [hunk 2]
```diff
@@ -25,13 +25,127 @@ defmodule Plausible.Release do
...
+      pending
+    end)
+  end
+
   def pending_migrations do
     prepare()
     IO.puts("Pending migrations")
     IO.puts("")
     Enum.each(repos(), &list_pending_migrations_for/1)
   end
```
Comment (zoldar) [inline lib/plausible_release.ex:117]: I think we can get rid of this function now as it's unused (along with private function used by it).

Comment (zoldar) [reply lib/plausible_release.ex:117]: Yes! 🔥 No point in keeping it around IMHO.

Comment (ruslandoga) [reply lib/plausible_release.ex]: https://github.com/plausible/analytics/pull/4466/commits/de576a363b971365ef9393fcacd871c8bbbe7596

**Review by zoldar — APPROVED** [review]

**Pull request merged**